src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -196,9 +206,17 @@ public IClient build() {
 			//if we need to really expose them.
 			dispatcher.setMaxRequests(maxRequests);
 			dispatcher.setMaxRequestsPerHost(maxRequestsPerHost);
+			final String userAgent = (this.userAgentPrefix != null ? this.userAgentPrefix + "/" : "") + "openshift-restclient-java/"+Version.userAgent();
```

Comment:
There are some useful utilities from the Apache deps StringUtils Objectutils which could make this prettier.